File: legacy/src/SoftielRemote.App.Flutter/lib/models/webrtc_signaling_message.dart
/// WebRTC signaling message model
class WebRTCSignalingMessage {
  final String type; // 'offer', 'answer', 'ice-candidate'
  final String targetDeviceId;
  final String senderDeviceId;
  final String connectionId;
  final String? sdp;
  final IceCandidate? iceCandidate;
  
  WebRTCSignalingMessage({
    required this.type,
    required this.targetDeviceId,
    required this.senderDeviceId,
    required this.connectionId,
    this.sdp,
    this.iceCandidate,
  });
  
  factory WebRTCSignalingMessage.fromJson(Map<String, dynamic> json) {
    return WebRTCSignalingMessage(
      type: json['type'] as String? ?? '',
      targetDeviceId: json['targetDeviceId'] as String? ?? '',
      senderDeviceId: json['senderDeviceId'] as String? ?? '',
      connectionId: json['connectionId'] as String? ?? '',
      sdp: json['sdp'] as String?,
      iceCandidate: json['iceCandidate'] != null
          ? IceCandidate.fromJson(json['iceCandidate'] as Map<String, dynamic>)
          : null,
    );
  }
  
  Map<String, dynamic> toJson() {
    return {
      'type': type,
      'targetDeviceId': targetDeviceId,
      'senderDeviceId': senderDeviceId,
      'connectionId': connectionId,
      if (sdp != null) 'sdp': sdp,
      if (iceCandidate != null) 'iceCandidate': iceCandidate!.toJson(),
    };
  }
}

/// ICE candidate model
class IceCandidate {
  final String candidate;
  final int sdpMLineIndex;
  final String? sdpMid;
  
  IceCandidate({
    required this.candidate,
    required this.sdpMLineIndex,
    this.sdpMid,
  });
  
  factory IceCandidate.fromJson(Map<String, dynamic> json) {
    return IceCandidate(
      candidate: json['candidate'] as String? ?? '',
      sdpMLineIndex: json['sdpMLineIndex'] as int? ?? 0,
      sdpMid: json['sdpMid'] as String?,
    );
  }
  
  Map<String, dynamic> toJson() {
    return {
      'candidate': candidate,
      'sdpMLineIndex': sdpMLineIndex,
      if (sdpMid != null) 'sdpMid': sdpMid,
    };
  }
}

File: legacy/src/SoftielRemote.Core/Dtos/WebRTCSignalingMessage.cs
using System.ComponentModel.DataAnnotations;

namespace SoftielRemote.Core.Dtos;

/// <summary>
/// WebRTC signaling mesajı (SDP offer/answer veya ICE candidate).
/// </summary>
public class WebRTCSignalingMessage
{
    /// <summary>
    /// Mesaj tipi (offer, answer, ice-candidate).
    /// </summary>
    [Required(ErrorMessage = "Type gereklidir")]
    [MaxLength(50, ErrorMessage = "Type maksimum 50 karakter olabilir")]
    [RegularExpression(@"^(offer|answer|ice-candidate)$", ErrorMessage = "Type 'offer', 'answer' veya 'ice-candidate' olmalıdır")]
    public string Type { get; set; } = string.Empty;

    /// <summary>
    /// Hedef Device ID (mesajın gönderileceği Agent veya Controller).
    /// </summary>
    [MaxLength(50, ErrorMessage = "TargetDeviceId maksimum 50 karakter olabilir")]
    [RegularExpression(@"^[0-9]{1,50}$", ErrorMessage = "TargetDeviceId sadece rakamlardan oluşmalıdır")]
    public string TargetDeviceId { get; set; } = string.Empty;

    /// <summary>
    /// Gönderen Device ID (mesajı gönderen Agent veya Controller).
    /// </summary>
    [MaxLength(50, ErrorMessage = "SenderDeviceId maksimum 50 karakter olabilir")]
    [RegularExpression(@"^[0-9]{1,50}$", ErrorMessage = "SenderDeviceId sadece rakamlardan oluşmalıdır")]
    public string SenderDeviceId { get; set; } = string.Empty;

    /// <summary>
    /// Connection ID (bağlantı isteğinin ID'si).
    /// </summary>
    [MaxLength(100, ErrorMessage = "ConnectionId maksimum 100 karakter olabilir")]
    public string ConnectionId { get; set; } = string.Empty;

    /// <summary>
    /// SDP offer veya answer (Type = "offer" veya "answer" ise).
    /// </summary>
    [MaxLength(10000, ErrorMessage = "SDP maksimum 10000 karakter olabilir")]
    public string? Sdp { get; set; }

    /// <summary>
    /// ICE candidate (Type = "ice-candidate" ise).
    /// </summary>
    public IceCandidateDto? IceCandidate { get; set; }
}

/// <summary>
/// ICE candidate bilgisi.
/// </summary>
public class IceCandidateDto
{
    /// <summary>
    /// ICE candidate string.
    /// </summary>
    [Required(ErrorMessage = "Candidate gereklidir")]
    [MaxLength(1000, ErrorMessage = "Candidate maksimum 1000 karakter olabilir")]
    public string Candidate { get; set; } = string.Empty;

    /// <summary>
    /// SDP media line index.
    /// </summary>
    [Range(0, 65535, ErrorMessage = "SdpMLineIndex 0 ile 65535 arasında olmalıdır")]
    public int SdpMLineIndex { get; set; }

    /// <summary>
    /// SDP media line ID.
    /// </summary>
    [MaxLength(50, ErrorMessage = "SdpMid maksimum 50 karakter olabilir")]
    public string? SdpMid { get; set; }
}



